### PR TITLE
soc: nrf52: Kconfig option for nRF52840 anomaly 198

### PR DESCRIPTION
**Pull request by rettichschnidi**
Enabling by default (if SPI3 used) because it affects all revisions
since "Engineering B", including the most recent one as of today
(revision 3).

Size changes when enabled:
 - -Og: flash +160 bytes (+0.02%), RAM +8 bytes (+0.01%)
 - -Os: flash +144 bytes (+0.02%), no change to RAM usage

### CODE REVIEW

File: soc/arm/nordic_nrf/nrf52/Kconfig.soc [hunk 1]
```diff
@@ -118,3 +118,11 @@ config NRF52_ANOMALY_132_DELAY_US
 	  depending on application, workaround may also need to be applied.
 	  Additional drivers initialization increases initialization time and delay
 	  may be shortened. Workaround is disabled by setting delay to 0.
+
+config NRF52_ANOMALY_198_WORKAROUND
+	bool "Anomaly 129 workaround"
```
Comment (koalatux) [inline soc/arm/nordic_nrf/nrf52/Kconfig.soc:123]:
```suggestion
	bool "Anomaly 198 workaround"
```

Comment (rettichschnidi) [reply soc/arm/nordic_nrf/nrf52/Kconfig.soc]: Fixed

**Review by koalatux — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

**Review by nordic-krch — DISMISSED** [review]

File: soc/arm/nordic_nrf/nrf52/Kconfig.soc [hunk 2]
```diff
@@ -118,3 +118,11 @@ config NRF52_ANOMALY_132_DELAY_US
 	  depending on application, workaround may also need to be applied.
 	  Additional drivers initialization increases initialization time and delay
 	  may be shortened. Workaround is disabled by setting delay to 0.
+
+config NRF52_ANOMALY_198_WORKAROUND
+	bool "Anomaly 198 workaround"
+	default y
+	depends on SOC_NRF52840
```
Comment (nordic-krch) [inline soc/arm/nordic_nrf/nrf52/Kconfig.soc:125]: what about having `depends on NRFX_SPIM3` as well?

Comment (rettichschnidi) [reply soc/arm/nordic_nrf/nrf52/Kconfig.soc:125]: Sounds sensible. Just added it.

**Review by anangl — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**